lib/MySqlDatabaseManager.js

```diff
@@ -1,16 +1,185 @@
 var DatabaseManager = require('./DatabaseManager').default
-  , classUtils = require('./class-utils');
+  , classUtils = require('./class-utils')
+  , mysql      = require('mysql')
+  , Promise = require('bluebird')
+  , _ = require('lodash');
 
 /**
  * @constructor
  * @extends DatabaseManager
+ *
+ * Notes:
+ *   - Even though the method signature implicates that _masterConnectionUrl returns
+ *     an URL string, it actually returns an object because MySQL node lib
+ *     assumes that the database name is defined in the URL format.
+ *
  */
 function MySqlDatabaseManager() {
   DatabaseManager.apply(this, arguments);
+  this._masterClient = null;
+  this._cachedTableNames = null;
 }
 
 classUtils.inherits(MySqlDatabaseManager, DatabaseManager);
 
+/**
+ * @Override
+ */
+MySqlDatabaseManager.prototype.createDbOwnerIfNotExist = function() {
+  return this._masterQuery("CREATE USER IF NOT EXISTS ?@'localhost' IDENTIFIED BY ?", [this.config.knex.connection.user, this.config.knex.connection.password])
+};
+
+/**
+ * @Override
+ */
+MySqlDatabaseManager.prototype.createDb = function(databaseName) {
+  databaseName = databaseName || this.config.knex.connection.database;
+  var collate = this.config.dbManager.collate;
+  var self = this;
+  var promise = Promise.reject();
+
+  if (_.isEmpty(collate)) {
+    promise = promise.catch(function () {
+      return self._masterQuery("CREATE DATABASE ?? DEFAULT CHARACTER SET utf8 DEFAULT COLLATE utf8_general_ci", [databaseName]);
+    });
+  } else {
+    // Try to create with each collate. Use the first one that works. This is kind of a hack
+    // but seems to be the only reliable way to make this work with both windows and unix.
+    _.each(collate, function(locale) {
+      promise = promise.catch(function() {
+        return self._masterQuery("CREATE DATABASE ?? DEFAULT CHARACTER SET utf8 DEFAULT COLLATE ?", [databaseName, locale]);
+      });
+    });
+  }
+
+  return promise;
+};
+
+/**
+ * Drops database with name if db exists.
+ *
+ * @Override
+ */
+MySqlDatabaseManager.prototype.dropDb = function(databaseName) {
+  databaseName = databaseName || this.config.knex.connection.database;
+  return this._masterQuery("DROP DATABASE IF EXISTS ??", [databaseName]);
+};
+
+/**
+ * @Override
+ */
+MySqlDatabaseManager.prototype.truncateDb = function(ignoreTables) {
+  var knex = this.knexInstance();
+  var config = this.config;
+
+  if (!this._cachedTableNames) {
+    this._updateTableNameCache(knex, config);
+  }
+
+  return this._cachedTableNames.then(function (tableNames) {
+    if (!_.isEmpty(tableNames)) {
+      return knex.transaction(function(trx) {
+        return knex.raw('SET FOREIGN_KEY_CHECKS = 0').transacting(trx)
+          .then(function() {
+            return Promise.map(tableNames, function(tableName) {
+              return knex.raw('TRUNCATE TABLE ' + tableName).transacting(trx);
```

Comment:
does this also reset the sequences? IIRC with knex one can do it like this `trx(tableName).truncate()`.